arch/RISCV/riscv_decode_compressed.gen.inc

```diff
@@ -0,0 +1,1043 @@
+/*=======================================================================*/
+/*This code was generated by the tool riscv_disasm_from_sail (see
+ * https://github.com/moste00/riscv_disasm_from_sail)*/
+/*from the sail model of RISC-V (see https://github.com/riscv/sail-riscv) @
+ * version 0b9c639f19da48734cbf14b61f6ad200b7c70de9.*/
+/*DO NOT MODIFY THIS CODE MANUALLY.*/
+/* */
+/*SPDX-License-Identifier: BSD-3-Clause*/
+/*=======================================================================*/
+
+#ifndef __Riscv_decode_compressed_gen_inc__
+#define __Riscv_decode_compressed_gen_inc__
+#include <stdint.h>
+
+#include <stddef.h>
+
+#include "riscv_ast.gen.inc"
+#include <string.h>
+
+void decode_compressed(struct ast *tree, uint64_t binary_stream) {
+  // ---------------------------C_NOP-------------------------------
+  {
```

Comment:
Unnecessary additional `{}` braces

Reply:
Oh those are necessary, because many of the If conditions and the associated context use variables with the same identifier. This wasn't a problem in the original Sail because every rule match forms a different scope, but it's a problem now in C.

The single easiest solution is adding those braces to start a new scope for each rule. I guess I could also solve the problem by making a new function `decode_<instruction_case>` to achieve the same effect, but that would complicate the generator a bit.